### PR TITLE
chore(sematext sink): Simplify the `host` calculation logic

### PR DESCRIPTION
**Pull request by bruceg**
The calculation of the `host` variable could use some clean up to make it clearer what is going on.

Signed-off-by: Bruce Guenter <bruce@timber.io>

### CODE REVIEW

**Comment by LucioFranco** [comment]
@bruceg I'm not sure this is actually clearer, I find I prefered the first method because it was easy to reason what was happening. First, we check our region setting, then if we have a custom host we override it, then we check if either were set to error out. 

I didn't go with the nested branches because I find that much harder to read personally.

File: src/sinks/sematext.rs [hunk 1]
```diff
@@ -35,28 +35,21 @@ pub enum Region {
...
-        }
-
-        // Test workaround for settings a custom host so we can test the body manually
-        if let Some(h) = &self.host {
-            host = Some(h.clone());
-        }
-
-        if host.is_none() {
+        let host = if let Some(h) = &self.host {
+            // Test workaround for settings a custom host so we can test the body manually
```
Comment (LucioFranco) [inline src/sinks/sematext.rs]: Actually, I forgot to remove this comment, its no longer jus used for testing.

**Comment by binarylogic** [comment]
Yeah, I appreciate the effort @bruceg. I'm a big 👍 on cleaning up things as you see them. If there is no functional change let's close this since @LucioFranco is the code owner here.

**Comment by LucioFranco** [comment]
Agreed @bruceg thanks for looking out :)

**Comment by bruceg** [comment]
Yeah, I was thinking we need a second (third) opinion.

**Comment by binarylogic** [comment]
I will summon @lukesteensen, the master tie breaker.

**Comment by lukesteensen** [comment]
This is cleaner than the code it's replacing, at least in my opinion. I'm almost never in favor of the style where you `let mut foo = None` followed by a bunch of procedural logic and then an `unwrap` at the end.

That being said, there's actually a third option I prefer that's even more explicit, shorter, and lets the user know when they've incorrectly set both options. I'll open a quick PR for that :smile:

**Pull request closed without merging**